app/src/mouse/input_processor_transform.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_input_processor_transform
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <drivers/input_processor.h>
+
+#include <dt-bindings/zmk/input_transform.h>
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/keymap.h>
+
+struct ipt_config {
+    size_t x_codes_size;
+    size_t y_codes_size;
+    uint8_t type;
+
+    const uint16_t *x_codes;
+    const uint16_t *y_codes;
+};
+
+static int code_idx(uint16_t code, const uint16_t *list, size_t len) {
+    for (int i = 0; i < len; i++) {
+        if (list[0] == code) {
```

Comment:
This should be comparing the list entry at the index
```suggestion
        if (list[i] == code) {
```